include/effects.h

```diff
@@ -106,81 +106,33 @@
-#define EFFECT_MATRIX_SMRAINBOW_TUNNEL                 167
-#define EFFECT_MATRIX_SMHYPNOSIS                       168
-#define EFFECT_MATRIX_SMWALKING_MACHINE                169
-#define EFFECT_MATRIX_SMXOR_CIRCLES                    170
-#define EFFECT_MATRIX_SMTIXY_LAND                      171
-#define EFFECT_MATRIX_SMBOID_EXPLOSION                 172
-#define EFFECT_MATRIX_SMSUPERNOVA                      173
-#define EFFECT_MATRIX_SMBLURRING_COLORS                174
-#define EFFECT_MATRIX_SMFLOW_FIELDS                    175
-#define EFFECT_MATRIX_SMGAMMA                          176
-#define EFFECT_MATRIX_SMSTARSHIP_TROOPERS              177
-#define EFFECT_MATRIX_SMHOLIDAY_LIGHTS                 178
-#define EFFECT_MATRIX_SMGOOGLE_NEXUS                   179
-#define EFFECT_MATRIX_SMGRAVITY_BALLS                  180
-#define EFFECT_MATRIX_SMFLOCKING                       181
-#define EFFECT_MATRIX_SMBUBBLES                        182
-#define EFFECT_MATRIX_SMPATTERN_TRICK                  183
-#define EFFECT_MATRIX_SMSUN_RADIATION                  184
-#define EFFECT_MATRIX_SMSIN_DOTS                       185
-#define EFFECT_MATRIX_SMFLYING                         186
-#define EFFECT_MATRIX_SMMETA_BALLS                     187
-#define EFFECT_MATRIX_SMTWISTER                        188
-#define EFFECT_MATRIX_SMTWIST                          189
-#define EFFECT_MATRIX_SMSPIRO_PULSE                    190
-#define EFFECT_MATRIX_SMCOLOR_POPCORN                  191
-#define EFFECT_MATRIX_SMPSP_CLOUD                      192
-#define EFFECT_MATRIX_SMEYE_TUNNEL                     193
-#define EFFECT_MATRIX_SMLIGHTNING                      194
+#define EFFECT_MATRIX_SM2DDPR                          137
+#define EFFECT_MATRIX_SMAMBERRAIN                      138
+#define EFFECT_MATRIX_SMBLURRING_COLORS                139
+#define EFFECT_MATRIX_SMFIRE2021                       140
+#define EFFECT_MATRIX_SMFLOW_FIELDS                    141
+#define EFFECT_MATRIX_SMGAMMA                          142
+#define EFFECT_MATRIX_SMHOLIDAY_LIGHTS                 143
+#define EFFECT_MATRIX_SMHYPNOSIS                       144
+#define EFFECT_MATRIX_SMMETA_BALLS                     145
+#define EFFECT_MATRIX_SMNOISE                          146
+#define EFFECT_MATRIX_SMPICASSO3IN1                    147
+#define EFFECT_MATRIX_SMRADIAL_FIRE                    148
+#define EFFECT_MATRIX_SMRADIAL_WAVE                    149
+#define EFFECT_MATRIX_SMRAINBOW_TUNNEL                 150
+#define EFFECT_MATRIX_SMSMOKE                          151
+#define EFFECT_MATRIX_SMSPIRO_PULSE                    152
+#define EFFECT_MATRIX_SMSTARDEEP                       153
+#define EFFECT_MATRIX_SMSTROBE_DIFFUSION               154
+#define EFFECT_MATRIX_SMSUPERNOVA                      155
+#define EFFECT_MATRIX_SMTIXY_LAND                      156
+#define EFFECT_MATRIX_SMTWISTER                        157
+#define EFFECT_MATRIX_SMWALKING_MACHINE                158
```

Comment:
It may well be a lesson for all of us, because we've never had a PR of this nature (or, in terms of additional effects, size). In hindsight, one thing that may have made things more manageable is cut the one massive PR into multiple smaller ones, each containing a few (max 10?) effects. Feedback cycles can be shorter, and the earlier review cycles may well provide input for later intended PRs/effects.

Just for the record: the code for the other effects is not gone, as I created an "all-robertlipe-effects" branch in this repo that contains all the effects that were in the codebase before the final selection. It actually also includes a number of effects that were originally developed by Dave that have now been removed with the same rigour.

Reply:
I agree. And on that point I'll ask Dave if he wants to have a(nother) look as well. Whether the answer is yes, no or absent, we'll take it from there and bring this to a merge.

Reply:
Dave's response, and I quote: "As long as it runs and builds, and Robert has looked at it, that’s enough for me!"